staging/src/k8s.io/client-go/rest/client.go

```diff
@@ -244,7 +244,9 @@ func (c *RESTClient) Patch(pt types.PatchType) *Request {
 
 // Get begins a GET request. Short for c.Verb("GET").
 func (c *RESTClient) Get() *Request {
-	return c.Verb("GET")
+	// Default cache-control to "no-store, no-cache" to make sure we
+	// don't store sensitive information.
+	return c.Verb("GET").SetHeader("Cache-control", "no-store, no-cache")
```

Comment:
Once we have seen the caching RT working in production without any issues and are confident in it, and once the server provides the cache-control headers, I **think** we can remove much of this and always go through the cache round tripper + never set headers from the client side (needs to be validated).